sunpy/map/tests/test_compositemap.py

```diff
@@ -34,6 +48,13 @@ def test_type_of_arguments_composite_map(composite_test_map):
         sunpy.map.CompositeMap(23, composite=True)
     assert str(excinfo.value) == 'CompositeMap expects pre-constructed map objects.'
 
+def test_autoalign_composite_map(composite_test_map, composite_test_reproject_map):
+    assert isinstance(composite_test_map.plot()[0], mpl.image.AxesImage)
+    assert isinstance(composite_test_map.plot()[1], mpl.collections.QuadMesh)
+    # check when map projected on same wcs
+    assert isinstance(composite_test_reproject_map.plot()[0], mpl.image.AxesImage)
+    assert isinstance(composite_test_reproject_map.plot()[1], mpl.image.AxesImage)
```

Comment:
```suggestion

def test_autoalign_needed(composite_test_map):
    # The overplot will use pcolormesh if the WCSs do not match
    images = composite_test_map.plot()
    assert isinstance(images[0], mpl.image.AxesImage)
    assert isinstance(images[1], mpl.collections.QuadMesh)


def test_autoalign_not_needed(composite_test_map_prealigned):
    # The overplot will use imshow if the WCSs match
    images = composite_test_map_prealigned.plot()
    assert isinstance(images[0], mpl.image.AxesImage)
    assert isinstance(images[1], mpl.image.AxesImage)
```
Better to split this into two separate tests